backend.native/compiler/ir/backend.native/src/org/jetbrains/kotlin/backend/konan/objcexport/ObjCExport.kt

```diff
@@ -226,6 +231,48 @@ internal class ObjCExport(val context: Context) {
 
         file.writeBytes(contents.toString().toByteArray())
     }
+
+    // See https://bugs.swift.org/browse/SR-10177
+    private fun ObjCExportedInterface.generateWorkaroundForSwiftSR10177() {
+        // Code for all protocols from the header should get into the binary.
+        // Objective-C protocols ABI is complicated (consider e.g. undocumented extended type encoding),
+        // so the easiest way to achieve this (quickly) is to compile a stub by clang.
+
+        val protocolsStub = listOf(
+                "__attribute__((used)) static void __workaroundSwiftSR10177() {",
+                buildString {
+                    append("    ")
+                    generatedClasses.forEach {
+                        if (it.isInterface) {
+                            val protocolName = namer.getClassOrProtocolName(it).objCName
+                            append("@protocol($protocolName); ")
+                        }
+                    }
+                },
+                "}"
+        )
+
+        val source = createTempFile("protocols", ".m").deleteOnExit()
+        source.writeLines(headerLines + protocolsStub)
+
+        val bitcode = createTempFile("protocols", ".bc").deleteOnExit()
+
+        val clangCommand = context.config.clang.clangC(
+                source.absolutePath,
+                "-O2",
+                "-emit-llvm",
+                "-c", "-o", bitcode.absolutePath
+        )
+
+        val result = Command(clangCommand).getResult(withErrors = true)
+
+        if (result.exitCode == 0) {
+            context.llvm.additionalProducedBitcodeFiles += bitcode.absolutePath
+        } else {
+            // Note: ignoring compile errors intentionally.
```

Comment:
Somewhat not very consistent - if error is important, why we ignore it?

Reply:
1) Because the same error encountered when importing the framework is more debuggable, e.g. because it doesn't involve any temp files and occurs at the moment a user can expect it
2) To keep the old behaviour.